solarforecastarbiter/io/reference_observations/common.py

```diff
@@ -83,6 +85,11 @@ def existing_observations(api):
     return {obs.name: obs for obs in api.list_observations()}
 
 
+@lru_cache(maxsize=4)
+def existing_forecasts(api):
+    return {fx.name: fx for fx in api.list_forecasts()}
```

Comment:
maybe ok, but these keys are not necessarily unique, right?

Reply:
especially if code below needs to handle limited field lengths...

Reply:
not necessarily unique, but should be for the reference user. I'll adjust the truncated name to just truncate the site name instead of using the abbreviation and I think that will mostly handle any issues